### DISCUSSION TITLE
Feature thoughts.

### DISCUSSION THREAD

**Discussion by sporked-thm** [bonukai/MediaTracker#659] [Ideas]
Subject: Question and Feature Idea for MediaTracker

I've been using MediaTracker ever since I had to manually scroll through a 10-page list of shows in a simple CherryTree table (2/10 — better than writing it on paper, but I wouldn’t recommend it!). During this time while using MediaTracker, I’ve encountered a couple of questions and have also come up with an idea for improvement.

The Question: Is it possible to export the user and content database for re-import? This would be useful for reinstallations or synchronizing content across multiple systems. I've looked into the database file created by the project, and it seems like it could be doable. Any insights on this would be appreciated.

Feature Idea #1: Could we implement a priority flag to adjust the display order of shows based on user-defined importance? The idea is to have a system where shows in the "upcoming" list are displayed with a weighted priority (e.g., low to high or high to low), it can default to 5 or something if it's not defined yet, so nobody needs to retroactively destroy their collections to inherit the feature.

Here's a rough pseudocode outline on the sort idea

upcomingShows.sort((a, b) => a.weight - b.weight); // Sort shows based on priority (low to high)
upcomingShows.forEach(show => {
  display(show); // Display shows in the sorted order
});

This would allow users to prioritize shows in their feed based on personal preference or importance, making it easier to focus on what matters most.

Feature Idea #2: In the current system, when viewing the Title screen for a TV show, the only mechanism to modify the "seen" history applies to all episodes up to the selected one. This means selecting Episode 8 will increment the "seen" count for Episodes 1-7 and mark Episode 8 as seen for the first time.

New proposed function: Add a shortcut button next to the "Add to seen history" link called "Add recently watched episode" that increments the "seen" count by 1 for the selected Title, setting the timestamp to "Now". This would allow users to quickly mark a single, pr recent episode as "seen" without modifying previous episodes, improving flexibility in tracking viewing history.

**Comment by sporked-thm** [comment]
Anwering part of my own question here,
the database appears to be a formatted sqlite db by default,  but the structure is minimalist!  Is there a good document that references the structure somewhere?

It appears that I can select some data from it and get some things out, however that's where the luck ends.  Any example queries to derive the entire structure ... looking through the code, it seems to leverage KNEX.. time to look up a reference or two.   

Does anyone have a recommendation on where to stick an artful logger statement to get the entire query used on the frontend, or the knex left/right for the more complex queries.

**Comment by bonukai** [comment]
`wip`  branch (#569) has implemented import/export. For the current version you can open your SQLite database [DB Browser for SQLite](https://sqlitebrowser.org/) and export/import tables to CSV.

Individual episodes can be added to the seen history from the "episodes" page, and there's a "First unwatched episode: ..." button on the details page (click on the "✓"). `wip` branch has an additional button to add a single episode from the details page.

`wip` branch has a highly customisable home screen, you could create multiple "Upcoming" sections, each with it's own list (for different priorities)

![image](https://github.com/user-attachments/assets/201221e7-5231-4a9c-8d01-5c87492c9e4f)
